tests/storage-metadata

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+set -eux
+
+# Install dependencies.
+# Releases before 24.04 don't have a yq deb so use the snap instead.
+# shellcheck disable=SC1091
+. /etc/os-release
+if dpkg --compare-versions "${VERSION_ID}" ge 24.04; then
+    install_deps jq yq
+else
+    install_deps jq
+    waitSnapdSeed
+    snap install yq
+fi
+
+# Install LXD.
+install_lxd
+
+IMAGE="${TEST_IMG:-ubuntu-minimal-daily:24.04}"
+
+# Configure LXD.
+lxc network create lxdbr0
+lxc profile device add default eth0 nic network=lxdbr0
+
+poolDriver="zfs"
+poolName="vmpool-${poolDriver}"
+
+echo "==> Create storage pool using driver ${poolDriver}"
+lxc storage create "${poolName}" "${poolDriver}"
+lxc profile device add default root disk pool="${poolName}" path=/
+
+if ! hasNeededAPIExtension backup_metadata_version; then
+    echo "LXD version $(lxd version | awk '{print $1}') doesn't support the new backup format, skipping test"
+    # shellcheck disable=SC2034
+    FAIL=0
+    exit 0
+fi
+
+# Do a matrix test and check both LXD 5.0 and 5.21
+# 5.0 is using the old format both internally and for exports.
+# 5.21 is already using the new format internally but falls back to the old format for exports.
+for vers in "5.0" "5.21"; do
+    echo "==> Create VM with a ${vers} LXD server using the original (old) backup format"
+    lxc launch "${IMAGE}" v1 --vm -s "${poolName}" -c limits.cpu=4 -c limits.memory=4GiB -d root,size=20GiB
+    waitInstanceReady v1
+
+    # workaround for LP: #2104066
+    lxc exec v1 -- mkdir -p /etc/systemd/system/snapd.service.d
+    lxc file push - v1/etc/systemd/system/snapd.service.d/override.conf << EOF
+# Workaround for https://bugs.launchpad.net/snapd/+bug/2104066
+[Service]
+Environment=SNAPD_STANDBY_WAIT=1m
+EOF
+    lxc exec v1 -- systemctl daemon-reload
+    lxc exec v1 -- systemctl restart snapd.service
+
+    # shellcheck disable=SC3044 # Ignore "declare is undefined" shellcheck error.
+    lxc exec v1 -- sh -c "$(declare -f waitSnapdSeed); waitSnapdSeed"
+    lxc exec v1 -- snap install lxd --channel="${vers}/edge"
```

Comment:
hrm should we be hardcoding `edge` here? What about when i run the test suite for `*/candidate` pre-release?